test/webdriverio/test/test_setup.ts

```diff
@@ -119,8 +121,16 @@ export async function testSetup(
   playgroundUrl: string,
   wdioWaitTimeoutMs: number,
 ): Promise<webdriverio.Browser> {
+  // Reset back to default state between tests.
+  synchronizeCoreBlocklyRendering = true;
+
   if (!driver) {
     driver = await driverSetup(wdioWaitTimeoutMs);
+  } else if (process.env.CI) {
+    // If running in CI force a session reload to ensure no browser state can
+    // leak across test suites (since this can sometimes cause complex combined
+    // failures in CI).
+    await driver.reloadSession();
```

Comment:
Is this run between each test or between suites? The PR description says between suites but this I think this is run between each test, though I may be mistaken. Not sure which one you intended so just wanted to double check.